<launch>

  <node pkg="apriltags" type="apriltags" name="apriltags" output="screen">
    <param name="~default_tag_size" value="0.046" />

    <param name="~viewer" value="true" />
    <param name="~publish_detections_image" value="true" />

    <param name="~display_type" value="CUBE" />
    <param name="~marker_thickness" value="0.02" />
    
   
    <rosparam command="delete" param="tag_data" />
    <rosparam param="tag_data">
      "0": 
        size: 0.046
      "1": 
        size: 0.046
      "2":
        size: 0.046
    </rosparam>

    <remap from="~image" to="/kinect2/hd/image_color_rect"/>
    <remap from="~camera_info" to="/kinect2/hd/camera_info"/>
    <remap from="~marker_array" to="/apriltags/marker_array"/>
    <remap from="~detections" to="/apriltags/detections"/>
  </node>
  
  <!-- ompl or stomp --> 
  <arg name="pipeline" default="ompl" />   
  
  
  <!-- rviz or gazebo graphical interface -->
  <arg name="gazebo_gui" default="false"/>
  <arg name="rviz_gui" default="true"/> 
  
  <!-- for moveit: fake_controller:=true, gazebo_controller:= false
       for gazebo: fake_controller:=false, gazebo_controller:=true
       for robot:  fake_controller:=false, gazebo_controller:=false --> 
  
  <arg name="fake_controller" default="true" />
  <arg name="gazebo_controller" default="false" />
  
  <rosparam command="load" file="$(find itia_kinova)/config/planning_scene_config.yaml"/>
  <rosparam command="load" file="$(find itia_kinova)/config/planning_scene_objects_info.yaml"/>
  
  <include file="$(find itia_kinova)/launch/kinova_j2n6s300.launch">
    <arg name="pipeline" default="$(arg pipeline)" />   
    <arg name="gazebo_gui" default="$(arg gazebo_gui)"/> 
    <arg name="rviz_gui" default="$(arg rviz_gui)"/> 
    <arg name="fake_controller" default="$(arg fake_controller)" />   
    <arg name="gazebo_controller" default="$(arg gazebo_controller)" />
    </include>

  <node name="object_tracking" pkg="itia_kinova" type="object_tracking" output="screen" />
  
</launch>
